### PR TITLE
Planned enhancement 8

### PR DESCRIPTION
**Pull request by lithops-zty**
history command list

### CODE REVIEW

**Review by huangjiaxi1111 — APPROVED** [review]
👍

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -769,7 +769,7 @@ More importantly, the parser will remove the backslash at the end of parsing, so
 4. **Enforce realistic role assignment for contacts.** Currently, a contact can have multiple roles, such as both "Professor" and "Student".
 This is unrealistic, as an individual is typically either a student or a professor, but not both.
```
Comment (lithops-zty) [inline docs/DeveloperGuide.md:770]:
```suggestionual is typically either a student or a professor, but not both.
```

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -787,11 +787,17 @@ We plan to allow the user to view the contacts in a sorted order to find the con
...
 such as `team leader`, `best friend`, `financial office` etc, and the current validation method does not offer this level of flexibility. Therefore, we plan to loosen the restriction
 on tag creation input to allow space for word separation, so that user can create multi-word tag to contacts for easier management.
 <br><br>
+
+8. **Reflect the true redo/undo states of the address book in the history command list.** Currently, the history command list only shows all data changing commands executed by the user and does not precisely reflect the current and history states of the address book.
+   For example, if the user executes `delete 1`, then `delete 2`, then `undo`, then `delete 3`, the true history states are `[delete 1, delete 3]` since `delete 2` has been purged.
+   However, the history command list will only show all data changing commands executed so far, (refer to the screenshot below) which is not accurate.
+   Moreover, we plan to highlight the current state of the address book in the history command list, so that the user can easily identify the current state of the address book.
+   ![point7_screenshot](images/planned_enhancements_screenshots/Point8.png)
```
Comment (MAOXIONGKAI) [inline docs/DeveloperGuide.md:803]:
```suggestion
   ![point8_screenshot](images/planned_enhancements_screenshots/Point8.png)
```

**Review by MAOXIONGKAI — APPROVED** [review]
LGTM 👍

**Pull request merged**